src/swupd_lib/target_root.c

```diff
@@ -756,5 +814,26 @@ int target_root_remove_files(struct list *files)
 		progress_report(count, total);
 	}
 
+	dirs_iter = list_head(dirs_to_remove);
+	while (dirs_iter) {
+		file = dirs_iter->data;
+		dirs_iter = dirs_iter->next;
+		fullfile = sys_path_join("%s/%s", globals.path_prefix, file->filename);
+		ret = sys_rm(fullfile);
+		if (ret != 0) {
+			ret = move_to_timestamp(file->filename);
+			if (ret != 0) {
+				error("Could not move file to backup location: %s\n", file->filename);
+			}
+		}
+		if (ret != 0) {
```

Comment:
Squash this `if` into the previous?

Reply:
Ah actually this was a mistake, the above if should have been checking if the error was another ENOTEMPTY instead of any error. Turning this into an else if though.